storage_controller/src/service.rs

```diff
@@ -287,6 +288,33 @@ impl ServiceState {
             0,
         );
     }
+
+    fn become_leader(&mut self) {
+        self.leadership_status = LeadershipStatus::Leader;
+
+        let status = &crate::metrics::METRICS_REGISTRY
+            .metrics_group
+            .storage_controller_leadership_status;
+
+        status.set(
```

Comment:
nit: this multi-tag update feels like it could be a helper in metrics.rs

Reply:
https://github.com/neondatabase/neon/pull/8588/commits/1cac7937a24490284466b87b9a97e948058a2e7f